server/src/service-layer/controllers/PaymentController.ts

```diff
@@ -0,0 +1,139 @@
+import PricingService from "business-layer/services/PricingService"
+import StripeService from "business-layer/services/StripeService"
+import { AuthServiceClaims } from "business-layer/utils/AuthServiceClaims"
+import {
+  MembershipTypeValues,
+  CHECKOUT_TYPE_KEY,
+  CheckoutTypeValues,
+  MEMBERSHIP_TYPE_KEY
+} from "business-layer/utils/StripeProductMetadata"
+import UserDataService from "data-layer/services/UserDataService"
+import { SelfRequestModel } from "service-layer/request-models/UserRequests"
+import { MembershipPaymentResponse } from "service-layer/response-models/PaymentResponse"
+import {
+  Controller,
+  Get,
+  Route,
+  Request,
+  Security,
+  SuccessResponse
+} from "tsoa"
+
+@Route("payment")
+export class PaymentController extends Controller {
+  @SuccessResponse("200", "Session created")
+  @Security("jwt")
+  @Get("membership")
+  public async getMembershipPayment(
+    @Request() request: SelfRequestModel
+  ): Promise<MembershipPaymentResponse> {
+    try {
+      const { uid, email, customClaims } = request.user
+      if (customClaims[AuthServiceClaims.MEMBER]) {
+        // Can't pay for membership if already member
+        this.setStatus(409)
+        return { error: "Already a member" }
+      }
+
+      const stripeService = new StripeService()
+      const userDataService = new UserDataService()
+
+      const userData = await userDataService.getUserData(uid)
+
+      /**
+       * Generate customer id if required
+       */
+      let stripeCustomerId: string
+      if (!userData.stripe_id) {
+        const { id } = await stripeService.createNewUser(email, uid)
+        stripeCustomerId = id
+        await userDataService.editUserData(uid, { stripe_id: stripeCustomerId })
+      } else {
+        stripeCustomerId = userData.stripe_id
+        /**
+         * See if user already has active session
+         */
+        const activeSession = await stripeService.getActiveSessionForUser(
+          stripeCustomerId,
+          CheckoutTypeValues.MEMBERSHIP
+        )
+        if (activeSession) {
+          const { client_secret, metadata } = activeSession
+          this.setStatus(200)
+          return {
+            clientSecret: client_secret,
```

Comment:
We might want to rename this `clientStripeSecret` or something to indicate it is the stripe secret, maybe.